src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -655,10 +657,25 @@ class WorkspaceService(protected val userInfo: UserInfo, dataSource: SlickDataSo
   def evaluateExpression(workspaceName: WorkspaceName, entityType: String, entityName: String, expression: String): Future[PerRequestMessage] =
     dataSource.inTransaction { dataAccess =>
       withWorkspaceContextAndPermissions(workspaceName, WorkspaceAccessLevels.Read, dataAccess) { workspaceContext =>
-        new SlickExpressionEvaluator(dataAccess).evalFinalAttribute(workspaceContext, entityType, entityName, expression) match {
-          case Success(result) => result.map(r => RequestComplete(StatusCodes.OK, r.toSeq))
-          case Failure(regret) => {
-            throw new RawlsExceptionWithErrorReport(errorReport = ErrorReport(StatusCodes.BadRequest, "Unable to evaluate expression '${expression}' on ${entityType}/${entityName} in ${workspaceName}", ErrorReport(regret)))
+        withSingleEntityRec(entityType, entityName, workspaceContext, dataAccess) { entities =>
+          SlickExpressionEvaluator.withNewExpressionEvaluator(dataAccess, entities) { evaluator =>
```

Comment:
I don't like the fact that it has context, but I agree that this is a better approach now that it does.